src/app/modules/item/components/sub-skills/sub-skills.component.ts

```diff
@@ -32,21 +33,23 @@ export class SubSkillsComponent implements OnChanges, OnDestroy {
     switchMap(({ id, attemptId }) => this.getItemChildrenService.get(id, attemptId)),
     map(children =>
       children
-        .filter(child => isASkill(child))
         .map(child => {
           const res = bestAttemptFromResults(child.results);
           return {
             ...child,
             isLocked: !canCurrentUserViewContent(child),
             result: res === null ? undefined : {
               attemptId: res.attemptId,
+              validated: res.validated,
               score: res.scoreComputed,
             },
           };
         })
     ),
     mapToFetchState({ resetter: this.refresh$ }),
   );
+  skills$ = this.state$.pipe(readyData(), map(children => children.filter(child => isASkill(child))));
+  activities$ = this.state$.pipe(readyData(), map(children => children.filter(child => !isASkill(child))));
```

Comment:
At the end, you use in your template both `state$` and `skills$` (some for `activities$`) which are not independent but looks like when used.
I would suggest you add to your `state$` another map which bind the list of children into a `{ skills: children.filter(child => isASkill(child)), activities: ... }`. This way, you can directly use `children.skills` in your template, instead doing `async` on another-by-related observable.

Reply:
Fixed